### PR TITLE
Change application configurations. Add Spring data

### PR DESCRIPTION
**Pull request by ViacheslavKruhlov**

### CODE REVIEW

**Comment by VyacheslavKruhlov** [comment]
Test database configurations should be changed to H2

**Comment by VyacheslavKruhlov** [comment]
Remove copying project `war` artifact into tomcat/webapps

File: BudgetHelper/src/main/java/com/budgethelper/dao/ItemDao.java [hunk 1]
```diff
@@ -1,20 +1,9 @@
...
 
 import com.budgethelper.model.Item;
-import com.budgethelper.model.RegularItem;
-import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface ItemDao extends CrudDao<Item> {
+public interface ItemDao extends JpaRepository<Item, Long> {
 
-	List<Item> getAllItems();
```
Comment (AlexKruhlov) [inline BudgetHelper/src/main/java/com/budgethelper/dao/ItemDao.java]: Please delete some empty lines.

File: BudgetHelper/src/main/java/com/budgethelper/service/impl/CategoryServiceImpl.java [hunk 2]
```diff
@@ -3,36 +3,35 @@
...
-	@Autowired
-	private CategoryDao categoryDao;
+	private final CategoryDao categoryDao;
 
-	@Override
-	public void save(final Category category) {
-		this.categoryDao.add(category);
+	public CategoryServiceImpl(final CategoryDao categoryDao) {
+		this.categoryDao = categoryDao;
 	}
```
Comment (AlexKruhlov) [inline BudgetHelper/src/main/java/com/budgethelper/service/impl/CategoryServiceImpl.java]: Please use the common formating for this file.

File: BudgetHelper/src/test/resources/hibernate.properties [hunk 3]
```diff
@@ -2,4 +2,7 @@ jdbc.driverClassName=org.postgresql.Driver
 jdbc.url=jdbc:postgresql://localhost:5432/budgethelper_test
 jdbc.username=postgres
 jdbc.password=root
+
 hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+hibernate.show_sql=false
+hibernate.format_sql=true
```
Comment (AlexKruhlov) [inline BudgetHelper/src/test/resources/hibernate.properties]: Please, add an epty line to the end of this file. Pay attention that git system set some special simbol in such cases.

File: BudgetHelper/src/main/java/com/budgethelper/service/impl/CategoryServiceImpl.java (hunk 2 above)
Comment (VyacheslavKruhlov) [reply BudgetHelper/src/main/java/com/budgethelper/service/impl/CategoryServiceImpl.java]: I don't understand. Can you explain that?

File: BudgetHelper/src/main/java/com/budgethelper/dao/ItemDao.java (hunk 1 above)
Comment (VyacheslavKruhlov) [reply BudgetHelper/src/main/java/com/budgethelper/dao/ItemDao.java]: done

File: BudgetHelper/src/test/resources/hibernate.properties (hunk 3 above)
Comment (VyacheslavKruhlov) [reply BudgetHelper/src/test/resources/hibernate.properties]: done

File: BudgetHelper/src/main/java/com/budgethelper/service/impl/CategoryServiceImpl.java (hunk 2 above)
Comment (AlexKruhlov) [reply BudgetHelper/src/main/java/com/budgethelper/service/impl/CategoryServiceImpl.java]: Please, open this file in your environment and I'll see that some lines of it have redundant whitespaces.

Comment (VyacheslavKruhlov) [reply BudgetHelper/src/main/java/com/budgethelper/service/impl/CategoryServiceImpl.java]: done

**Pull request merged**